enhancements/network/user-defined-network-segmentation.md

```diff
@@ -379,11 +382,283 @@ any user-defined primary network.
+}
+
+// ClusterUserDefinedNetwork defines the desired state of ClusterUserDefinedNetwork.
+type ClusterUserDefinedNetwork struct {
+    // NamespaceSelector Label selector for which namespace network should be available for.
+    // +kubebuilder:validation:Required
+    // +required
+    NamespaceSelector metav1.LabelSelector `json:"namespaceSelector"`
+    
+    // Template is direct specification of UserDefinedNetwork.
+    // +kubebuilder:validation:Required
+    // +required
+    Template *UserDefinedNetworkTemplateSpec `json:"template"`
+}
+
+// UserDefinedNetworkTemplateSpec UserDefinedNetwork spec template.
+type UserDefinedNetworkTemplateSpec struct {
+    // UserDefinedNetworkSpec contains the UserDefinedNetwork specification.
+    Spec UserDefinedNetworkSpec `json:"spec,omitempty"`
+}
+
+// ClusterUserDefinedNetworkStatus contains the observed status of the ClusterUserDefinedNetwork.
+type ClusterUserDefinedNetworkStatus struct {
+    // ActiveNamespaces indicates in which namespaces network is available.
+    ActiveNamespaces []string `json:"activeNamespaces,omitempty"`
+    
+    Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+```
+
+Suggested CRD short-name: `cudn`
+
+###### Example
+Condition reflecting the network readiness:
+```yaml
+status:
+  conditions:
+  - type: NetworkReady
+    status: "True"
+    reason: NetworkAttachmentDefinitionReady
+    message: NetworkAttachmentDefinition has been created
+```
+Condition reflecting the network readiness underlying OVN network state:
+```yaml
+status:
+  conditions:
+  - type: NetworkCreated
+    status: "True"
+    reason: OVNNetworkCreated
+    message: OVN network has been created
+```
+
+##### Example - Namespace scoped network
+```yaml
+kind: UserDefinedNetwork
+metadata:
+  name: db-network
+  namespace: demo
+spec:
+  topology: Layer2
+  role: Primary
+  mtu: 9000
+  subnets: ["10.0.0.0/24"]
+  excludeSubnets: ["10.0.0.0/26"]
+  ipamLifecycle: Persistent
+```
+
+After creation:
+```yaml
+kind: UserDefinedNetwork
+metadata:
+  name: db-network
+  namespace: demo
+  finalizers:
+    - k8s.ovn.org/user-defined-network-protection
+spec:
+  topology: Layer2
+  role: Primary
+  mtu: 9000
+  subnets: ["10.0.0.0/24"]
+  excludeSubnets: ["10.0.0.100/26"]
+  ipamLifecycle: Persistent
+status:
+  conditions:
+  - type: "NetworkReady"
+    status: "True"
+    reason: "NetworkAttachmentDefinitionReady"
+    message: "NetworkAttachmentDefinition has been created"
+```
+
+##### Example - Cluster scoped network
+```yaml
+kind: ClusterUserDefinedNetwork
+metadata:
+  name: db-network
+spec:
+  namespaceSelector:
+    matchExpressions:
+    - key: kubernetes.io/metadata.name
+      operator: In
+      values:
+        - "mynamespace"
+        - "theirnamespace"
+  template:
+    topology: Layer2
+    role: Primary
+    mtu: 9000
+    subnets: ["10.0.0.0/24"]
+    excludeSubnets: ["10.0.0.100/26"]
+status:
+  activeNamespaces:
```

Comment:
Following discussion with the team, currently there is no additional details we can think of.
It was suggested to change the `ActiveNamespce` field defintion to enable extending and avoiding bare string slice:
```
(pkg/crd/userdefinednetwork/types.go)
...
ActiveNamesapces []NamespaceStatus
...
type NamespaceStatus struct {
  // Name the target namesapce name
  Name string `json:"name"`
  // ActiveSince indicate since when the network is available.
  ActiveSince time.Date `json:"activeSince"`
}
(cluster UDN CR status example)
status:
  activeNamespaces:
  -  myNs: 
      activeSince:  <TS>
```